frontend/src/app/core/state/file-links/file-links.service.ts

```diff
@@ -90,4 +91,36 @@ export class FileLinksResourceService extends ResourceCollectionService<IFileLin
   protected createStore():CollectionStore<IFileLink> {
     return new FileLinksStore();
   }
+
+  remove(collectionKey:string, fileLink:IFileLink):void {
+    if (!fileLink._links.delete) {
+      return;
+    }
+
+    const headers = new HttpHeaders({ 'Content-Type': 'application/json' });
+    this.http
+      .delete<void>(fileLink._links.delete.href, { withCredentials: true, headers })
+      .pipe(
+        catchError((error) => {
+          this.toastService.addError(error);
+          throw error;
+        }),
+      )
+      .subscribe(() => {
+        applyTransaction(() => {
+          this.store.remove(fileLink.id);
+          this.store.update(({ collections }) => (
```

Comment:
We have a function for updating collections, havn't we?

Reply:
extracted to higher level function